### ISSUE TITLE
Missing key exception when raising StrayQuoteError

### ISSUE THREAD

**Issue by rockwood** [beatrichartz/csv#96]
Parsing an invalid CSV that contains stray quotes raises a KeyError instead of the expected StrayQuoteError. It appears that `yield_or_raise!/2` is not passing the correct arguments to the exception. 

![image](https://user-images.githubusercontent.com/659052/86988016-44377700-c1d2-11ea-8ae4-6c5eff18aa2a.png)

```
key :field not found in: [message: "=", line: 2]
```

```
 elixir lib/keyword.ex:399 Keyword.fetch!/2
 csv lib/csv/exceptions.ex:46 CSV.StrayQuoteError.exception/1
 csv lib/csv.ex:214 CSV.yield_or_raise!/2
 elixir lib/stream.ex:572 anonymous fn/4 in Stream.map/2
 elixir lib/enum.ex:3686 Enumerable.List.reduce/3
 elixir lib/stream.ex:931 Stream.do_list_transform/7
 elixir lib/stream.ex:1609 Enumerable.Stream.do_each/4
```

Let me know if I'm doing something wrong. I'm happy to work on a PR.

**Comment by beatrichartz** [comment]
Hey @rockwood thanks for raising this this is fixed in `2.4.0` - give it a go and let me know if you run into further issues.

**Comment by rockwood** [comment]
Thanks @beatrichartz!
